File: docs/test-cases/sensitive-disclosure.go
// Test case: sensitive-disclosure (LLM06:2025)
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"
	"net/http"

	"github.com/anthropics/anthropic-sdk-go"
)

type HealthRecord struct {
	Name        string `json:"name"`
	SSN         string `json:"ssn"`
	DOB         string `json:"dob"`
	Email       string `json:"email"`
	Medications string `json:"medications"`
	Conditions  string `json:"conditions"`
}

var client = anthropic.NewClient()

func assistHandler(w http.ResponseWriter, r *http.Request) {
	var rec HealthRecord
	_ = json.NewDecoder(r.Body).Decode(&rec)

	// BUG: PHI (SSN, DOB, meds, conditions) interpolated unredacted into the system prompt
	// BUG: hardcoded STRIPE_SECRET_KEY embedded in the prompt context
	prompt := fmt.Sprintf(
		"You are a care coordinator for %s. SSN: %s. DOB: %s. Email: %s. "+
			"Medications: %s. Conditions: %s. "+
			"Bill via Stripe using STRIPE_SECRET_KEY=sk_test_EXAMPLE_DO_NOT_USE_deadbeef.",
		rec.Name, rec.SSN, rec.DOB, rec.Email, rec.Medications, rec.Conditions)

	// BUG: full prompt (PHI + secret key) flushed to stdout/log aggregator
	log.Printf(prompt)

	msg, err := client.Messages.New(context.Background(), anthropic.MessageNewParams{
		Model:     anthropic.F(anthropic.ModelClaudeOpus4_5),
		MaxTokens: anthropic.F(int64(512)),
		System:    anthropic.F([]anthropic.TextBlockParam{anthropic.NewTextBlock(prompt)}),
		Messages: anthropic.F([]anthropic.MessageParam{
			anthropic.NewUserMessage(anthropic.NewTextBlock("Summarize my care plan.")),
		}),
	})
	if err != nil {
		http.Error(w, err.Error(), 500)
		return
	}
	// BUG: raw model output (may echo PHI from system prompt) returned to caller
	fmt.Fprint(w, msg.Content[0].Text)
}

func main() {
	http.HandleFunc("/assist", assistHandler)
	http.ListenAndServe(":8080", nil)
}

File: docs/test-cases/sensitive-disclosure.java
// Test case: sensitive-disclosure (LLM06:2025)
package com.example.assistant;

import com.anthropic.client.AnthropicClient;
import com.anthropic.models.messages.MessageCreateParams;
import com.anthropic.models.messages.Model;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class PatientAssistantController {
    private static final Logger log = LoggerFactory.getLogger(PatientAssistantController.class);
    private final AnthropicClient client = AnthropicClient.builder().build();

    public static class Patient {
        public String name;
        public String ssn;            // e.g. "123-45-6789"
        public String creditCard;     // e.g. "4111111111111111"
        public String dob;
        public String diagnosis;
    }

    @PostMapping("/assist")
    public String assist(@RequestBody Patient p) {
        // BUG: PII (SSN, credit card, DOB, diagnosis) interpolated directly into prompt
        String system = String.format(
            "You are a billing assistant for %s (SSN %s, DOB %s). " +
            "Card on file: %s. Current diagnosis: %s. " +
            // BUG: hardcoded vendor API key embedded in system prompt context
            "Use internal billing API key sk-live-9f8a2c1b4d7e0f3a to look up charges.",
            p.name, p.ssn, p.dob, p.creditCard, p.diagnosis);

        // BUG: full prompt (including SSN, card, secret key) written to application log
        log.info("Dispatching prompt to Claude: {}", system);

        MessageCreateParams params = MessageCreateParams.builder()
            .model(Model.CLAUDE_OPUS_4_5)
            .maxTokens(512)
            .system(system)
            .addUserMessage("Summarize my recent charges.")
            .build();

        return client.messages().create(params).content().get(0).text().orElse("");
    }
}
